bin/s2nc.c

```diff
@@ -229,6 +232,31 @@ static void setup_s2n_config(struct s2n_config *config, const char *cipher_prefs
     GUARD_EXIT(s2n_config_send_max_fragment_length(config, mfl_code), "Error setting maximum fragment length");
 }
 
+static void send_cstring(struct s2n_connection *conn, int sockfd, const char *data) {
+    unsigned long bytes_remaining = strlen(data);
+    const char *data_ptr = data;
+    s2n_blocked_status blocked;
+    do {
+        s2n_errno = S2N_ERR_T_OK;
+        ssize_t send_len = MIN(bytes_remaining, INT_MAX);
```

Comment:
Shouldn't "INT_MAX" be "SSIZE_MAX"? Why "INT_MAX"? If we can't use SSIZE_MAX for some reason, I'd rather we intentionally chose a well defined but smaller size, like UINT16_MAX.

Reply:
I didn't know there was an SSIZE_MAX, that definitely makes more sense!